plugin/PlayAssetDelivery/src/test/java/com/google/play/core/godot/assetpacks/PlayAssetDeliveryTestHelper.java

```diff
@@ -46,4 +55,55 @@ public static Dictionary createAssetPackLocationsDictionary() {
     testDict.put("location2", innerDict2);
     return testDict;
   }
+
+  /**
+   * Mock object factory that returns a mock Task<T> object. Will invoke onSuccessListener with
+   * result if addOnSuccessListener() is called.
+   *
+   * @param result object to be passed to the onSuccessListener
+   * @param <T> parameterized type of the result returned by this task if it succeeds
+   * @return instantiated mock Task
+   */
+  public static <T> Task<T> createMockOnSuccessTask(T result) {
+    Task<T> returnTaskMock = mock(Task.class);
+    doAnswer(
+            invocation -> {
+              OnSuccessListener<T> listener = (OnSuccessListener<T>) invocation.getArguments()[0];
+              listener.onSuccess(result);
+              return null;
+            })
+        .when(returnTaskMock)
+        .addOnSuccessListener(any(OnSuccessListener.class));
+    return returnTaskMock;
+  }
+
+  /**
+   * Mock object factory that returns a mock Task<T> object. Will invoke onFailureListener with
+   * onFailureException if addOnFailureListener() is called.
+   *
+   * @param onFailureException Exception to be passed to the onFailureListener
+   * @param <T> parameterized type of the result returned by this task if it succeeds
+   * @return instantiated mock Task
+   */
+  public static <T> Task<T> createMockOnFailureTask(Exception onFailureException) {
+    Task<T> returnTaskMock = mock(Task.class);
+    doAnswer(
+            invocation -> {
+              OnFailureListener listener = (OnFailureListener) invocation.getArguments()[0];
+              listener.onFailure(onFailureException);
+              return null;
+            })
+        .when(returnTaskMock)
+        .addOnFailureListener(any(OnFailureListener.class));
+    return returnTaskMock;
+  }
+
+  /** Returns a mock AssetPackException object used for unit testing. */
+  public static AssetPackException createMockAssetPackException() {
```

Comment:
Similar to createMockOnFailureTask, I would pass the toString() and error code values as arguments to this method.